src/core/worker.c

```diff
@@ -204,6 +204,7 @@ QuicWorkerAssignConnection(
 {
     CXPLAT_DBG_ASSERT(Connection->Worker != Worker);
     Connection->Worker = Worker;
+    Connection->Partition = Worker->Partition;
```

Comment:
nit: SAL annotations on this function are wrong, it should be `_Inout_` for the connection (and while at it, `const` for the worker).
But not directly tied to the PR.

Reply:
We can follow up in a separate PR to try to fix these.